File: infra/src/test/java/mixter/infra/SpyEventPublisher.java
package mixter.infra;

import mixter.domain.Event;
import mixter.domain.EventPublisher;

import java.util.ArrayList;
import java.util.List;

public class SpyEventPublisher implements EventPublisher {
    public List<Event> publishedEvents = new ArrayList<>();

    public void publish(Event event) {
        publishedEvents.add(event);
    }
}

File: infra/src/test/scala/mixter/infra/SpyEventPublisher.scala
package mixter.infra

import mixter.domain.{Event, EventPublisher}

class SpyEventPublisher extends EventPublisher {
  var publishedEvents:Seq[Event]=Seq.empty
  override def publish[T<:Event](event:T): Unit = {
    publishedEvents = publishedEvents :+ event
  }
}
